### PR TITLE
samples: cellular: nrf_cloud_rest_fota: add smp fota for nrf9160dk

### PR DESCRIPTION
**Pull request by jayteemo**
Add the option to enable SMP FOTA on the nRF9160DK. 
IRIS-9481

### CODE REVIEW

**Review by tony-le-24 — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -498,7 +498,11 @@ Cellular samples
 
 * :ref:`nrf_cloud_rest_fota` sample:
 
-  * Added support for setting the FOTA update check interval using the config section in the shadow.
+  * Added:
+
+    * Support for setting the FOTA update check interval using the config section in the shadow.
+    * Support for SMP FOTA on the nRF9160DK.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:504]:
```suggestion
    * Support for SMP FOTA on the nRF9160 DK.
```

File: samples/cellular/nrf_cloud_rest_fota/README.rst [hunk 2]
```diff
@@ -115,6 +115,17 @@ To enable full modem FOTA, add the following parameter to your build command:
...
+To enable SMP FOTA (nRF9160 DK only), add the following parameters to your build command:
+
+``-DEXTRA_CONF_FILE=overlay_smp_fota.conf -DEXTRA_DTC_OVERLAY_FILE=nrf9160dk_mcumgr_client_uart2.overlay``
+
+The nRF52840 device on your DK must be running an SMP compatible firmware, such as the :ref:`smp_svr` sample.
+Build the :ref:`smp_svr` sample for the ``nrf9160dk/nrf52840`` board with the following parameters:
+
+``-DEXTRA_CONF_FILE=overlay_smp_fota.conf -DEXTRA_DTC_OVERLAY_FILE=nrf9160dk_mcumgr_client_uart2.overlay``
+
+To change :ref:`smp_svr` sample's application version, set the ``CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION`` Kconfig option.
```
Comment (peknis) [inline samples/cellular/nrf_cloud_rest_fota/README.rst:127]:
```suggestion
To change :ref:`smp_svr` sample's application version, set the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option.
```

File: samples/cellular/nrf_cloud_rest_fota/README.rst [hunk 3]
```diff
@@ -115,6 +115,17 @@ To enable full modem FOTA, add the following parameter to your build command:
 Also, if you are using an nRF9160 DK, specify your development kit version by appending it to the board name.
 For example, if you are using version 1.0.1, use the board name ``nrf9160dk@1.0.1/nrf9160/ns`` in your build command.
 
+To enable SMP FOTA (nRF9160 DK only), add the following parameters to your build command:
+
+``-DEXTRA_CONF_FILE=overlay_smp_fota.conf -DEXTRA_DTC_OVERLAY_FILE=nrf9160dk_mcumgr_client_uart2.overlay``
```
Comment (peknis) [inline samples/cellular/nrf_cloud_rest_fota/README.rst:120]:
```suggestion
* ``-DEXTRA_CONF_FILE=overlay_smp_fota.conf``
* ``-DEXTRA_DTC_OVERLAY_FILE=nrf9160dk_mcumgr_client_uart2.overlay``
```

File: samples/cellular/nrf_cloud_rest_fota/README.rst [hunk 4]
```diff
@@ -115,6 +115,17 @@ To enable full modem FOTA, add the following parameter to your build command:
...
+
+The nRF52840 device on your DK must be running an SMP compatible firmware, such as the :ref:`smp_svr` sample.
+Build the :ref:`smp_svr` sample for the ``nrf9160dk/nrf52840`` board with the following parameters:
+
+``-DEXTRA_CONF_FILE=overlay_smp_fota.conf -DEXTRA_DTC_OVERLAY_FILE=nrf9160dk_mcumgr_client_uart2.overlay``
```
Comment (peknis) [inline samples/cellular/nrf_cloud_rest_fota/README.rst:125]:
```suggestion
* ``-DEXTRA_CONF_FILE=overlay_smp_fota.conf``
* ``-DEXTRA_DTC_OVERLAY_FILE=nrf9160dk_mcumgr_client_uart2.overlay``
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

File: samples/cellular/nrf_cloud_rest_fota/CMakeLists.txt [hunk 5]
```diff
@@ -12,4 +12,8 @@ zephyr_compile_definitions(PROJECT_NAME=${PROJECT_NAME})
 
 # NORDIC SDK APP START
 target_sources(app PRIVATE src/main.c)
+
+if(CONFIG_NRF_CLOUD_FOTA_SMP AND CONFIG_BOARD_NRF9160DK_NRF9160_NS)
+target_sources(app PRIVATE src/smp_reset.c)
```
Comment (nordicjm) [inline samples/cellular/nrf_cloud_rest_fota/CMakeLists.txt:17]: 2 space indent for cmake

File: samples/cellular/nrf_cloud_rest_fota/CMakeLists.txt [hunk 6]
```diff
@@ -12,4 +12,8 @@ zephyr_compile_definitions(PROJECT_NAME=${PROJECT_NAME})
 
 # NORDIC SDK APP START
 target_sources(app PRIVATE src/main.c)
+
+if(CONFIG_NRF_CLOUD_FOTA_SMP AND CONFIG_BOARD_NRF9160DK_NRF9160_NS)
```
Comment (nordicjm) [inline samples/cellular/nrf_cloud_rest_fota/CMakeLists.txt:16]: is this specifically tied to a board or is it just the non-secure core? I.e. how can customers use this on their own boards?

File: samples/cellular/nrf_cloud_rest_fota/src/smp_reset.h [hunk 7]
```diff
@@ -0,0 +1,20 @@
+/* Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _SMP_RESET_H_
+#define _SMP_RESET_H_
+
+#if defined(CONFIG_NRF_CLOUD_FOTA_SMP) && defined(CONFIG_BOARD_NRF9160DK_NRF9160_NS)
```
Comment (nordicjm) [inline samples/cellular/nrf_cloud_rest_fota/src/smp_reset.h:9]: Likewise with above

**Review by nordicjm — COMMENTED** [review]
Some questions before approving (and a nit.)

File: samples/cellular/nrf_cloud_rest_fota/CMakeLists.txt (hunk 6 above)
Comment (jayteemo) [reply samples/cellular/nrf_cloud_rest_fota/CMakeLists.txt:16]: for this sample we are only providing an example for this (`nrf9160dk/nrf9160/ns`) build target.
customers would need to modify it to work with other boards.

**Review by nordicjm — APPROVED** [review]

**Comment by rlubos** [comment]
@jayteemo We've missed RC1, but this could still go in, just need to align the PR a bit. Please drop release notes change, and post a comment in aggregate PR https://github.com/nrfconnect/sdk-nrf/pull/18192 instead,

**Comment by jayteemo** [comment]
@rlubos thanks, [done](https://github.com/nrfconnect/sdk-nrf/pull/18192/files#r1817034872)

**Pull request merged**